packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts

```diff
@@ -271,30 +269,50 @@ export class ChainAdapter implements IChainAdapter<ChainTypes.Bitcoin> {
       throw new Error('undefined fee')
 
     // We have to round because coinselect library uses sats per byte which cant be decimals
-    const fast = String(Math.round(feeData.data.fast?.satsPerKiloByte / 1024))
-    const average = String(Math.round(feeData.data.average?.satsPerKiloByte / 1024))
-    const slow = String(Math.round(feeData.data.slow?.satsPerKiloByte / 1024))
+    const fastPerByte = String(Math.round(feeData.data.fast?.satsPerKiloByte / 1024))
+    const averagePerByte = String(Math.round(feeData.data.average?.satsPerKiloByte / 1024))
+    const slowPerByte = String(Math.round(feeData.data.slow?.satsPerKiloByte / 1024))
+
+    const { data: utxos } = await this.providers.http.getUtxos({
+      pubkey
+    })
+
+    type MappedUtxos = Omit<bitcoin.api.Utxo, 'value'> & { value: number }
+    const mappedUtxos: MappedUtxos[] = utxos.map((x) => ({ ...x, value: Number(x.value) }))
+
+    const { fee: fastFee } = coinSelect<MappedUtxos, chainAdapters.bitcoin.Recipient>(
+      mappedUtxos,
+      [{ value: Number(value), address: to }],
+      Number(fastPerByte)
+    )
+    const { fee: averageFee } = coinSelect<MappedUtxos, chainAdapters.bitcoin.Recipient>(
+      mappedUtxos,
+      [{ value: Number(value), address: to }],
+      Number(averagePerByte)
+    )
+    const { fee: slowFee } = coinSelect<MappedUtxos, chainAdapters.bitcoin.Recipient>(
+      mappedUtxos,
+      [{ value: Number(value), address: to }],
+      Number(slowPerByte)
+    )
 
     const confTimes: chainAdapters.FeeDataEstimate<ChainTypes.Bitcoin> = {
       [chainAdapters.FeeDataKey.Fast]: {
-        feePerUnit: fast,
+        txFee: String(fastFee),
```

Comment:
Has this value been checked to be a valid numeric value first?

Reply:
Numeric value being returned from `coinselect` package.

Reply:
should be good here from coinselect number value

Reply:
and coinselect has a strong guarantee on the return value?

Reply:
Sorry i didnt see this until after I merged. I can follow up with a small pr checking that it is a defined number